src/nni_manager/training_service/kubernetes/frameworkcontroller/frameworkcontrollerTrainingService.ts

```diff
@@ -451,13 +453,30 @@ class FrameworkControllerTrainingService extends KubernetesTrainingService imple
                     mountPath: '/mnt/frameworkbarrier'
                 }]
         }];
-        const spec: any = {
+        let spec: any;
+        if(privateRegistrySecretName) {
+            spec = {
                 containers: containers,
+                imagePullSecrets: [
+                   {
+                       name: privateRegistrySecretName 
+                   }
+                ],
                 initContainers: initContainers,
                 restartPolicy: 'OnFailure',
                 volumes: volumeSpecMap.get('nniVolumes'),
                 hostNetwork: false
-        };
+            };
+        } else {
+            spec = {
+                containers: containers,
+                initContainers: initContainers,
+                restartPolicy: 'OnFailure',
```

Comment:
Duplicated code,  it is suggested to eliminate duplication by reusing the common part.

Reply:
fixed.